src/java/org/apache/cassandra/repair/RepairProgressReporter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.cassandra.repair;
+
+import java.util.Collection;
+import java.util.UUID;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.dht.Range;
+import org.apache.cassandra.dht.Token;
+import org.apache.cassandra.repair.messages.RepairOption;
+
+public interface RepairProgressReporter
+{
+    RepairProgressReporter instance = CassandraRelevantProperties.REPAIR_PROGRESS_REPORTER.isPresent()
+                                      ? make(CassandraRelevantProperties.REPAIR_PROGRESS_REPORTER.getString())
+                                      : new DefaultRepairProgressReporter();
+
+    void startParentRepair(UUID parentSession, String keyspaceName, String[] cfnames, RepairOption options);
+
+    void successfulParentRepair(UUID parentSession, Collection<Range<Token>> successfulRanges);
+
+    void failParentRepair(UUID parentSession, Throwable t);
```

Comment:
is it a spelling issue, or is a `Reporter` able to `fail` a repair?

Reply:
if the latter, then it's not a simple reporter, but can perform actions too... perhaps some javadoc would be in order?

Reply:
(same applies to `startRepairs` and `failRepairs`)

Reply:
I will rename those methods to `onXXX`